app/src/main/java/org/asteroidos/sync/services/NLService.java

```diff
@@ -227,9 +228,16 @@ public void onReceive(Context context, Intent intent) {
             if (intent.getStringExtra("command").equals("refresh")) {
                 Handler handler = new Handler();
                 handler.postDelayed(() -> {
-                    while (!listenerConnected);
+                    while (!listenerConnected) {
+                        // Sleep the spin
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                            Thread.onSpinWait();
+                        }else {
```

Comment:
The else case is a bit useless here.

Reply:
It is, I think I left it  there just in case someone figures out how to handle spin waits on a older android systems.